### DISCUSSION TITLE
https://theme-hope.vuejs.press/zh/guide/component/global.html

### DISCUSSION THREAD

**Discussion by giscus** [vuepress-theme-hope/giscus-discussions#931] [Announcements]
# https://theme-hope.vuejs.press/zh/guide/component/global.html

本教程将指引你如何在 VuePress 项目中注册全局组件。

https://theme-hope.vuejs.press/zh/guide/component/global.html

**Comment by lumengde123** [comment]
这里的方法二：**通过客户端配置文件注册** 反馈一下，`import MyComponent from "./MyComponent.vue";`这句即使路径填对了也会报错，原因是ts文件识别不出vue类型，需要处理一下

**Comment by lumengde123** [comment]
在`.vuepress/`文件下新建`env.d.ts`文件，填入：
```typescript
declare module '*.vue' {
    import type { DefineComponent } from 'vue';
    const vueComponent: DefineComponent<{}, {}, any>;
    export default vueComponent;
}```
纯小白，折腾半天，分享一下...(评论框内`shift`+`Enter`可以换行)
